lib/matplotlib/backend_tools.py

```diff
@@ -1017,6 +1017,47 @@ def _mouse_move(self, event):
         self.toolmanager.canvas.draw_idle()
 
 
+class HelpTool(ToolToggleBase):
+    description = 'Print tool list and shortcuts'
+    default_keymap = rcParams['keymap.help']
+    image = 'help.png'
+
+    def __init__(self, *args):
+        ToolToggleBase.__init__(self, *args)
+        self.text_axes = None
+
+    def enable(self, *args):
+        self.text_axes = self.figure.add_axes((0, 0, 1, 1))
+        t = self.get_text()
+        self.text_axes.text(0., 1., t,
+                            va='top',
+                            family='monospace',
+                            backgroundcolor='w',
+                            alpha=1.0)
+        self.figure.canvas.draw_idle()
+
+    def disable(self, *args):
+        self.text_axes.remove()
+        self.figure.canvas.draw_idle()
+
+    def get_text(self):
+        sep = '_' * 80 + '\n'
+        t = sep
+        t += "{0:15} {1:25} {2}\n".format(
```

Comment:
*I am no expert of fonts*, but I think there is no way to *precisely* know that in a *simple* way (i.e. outside of the "font engine"). However some [SO sources](https://stackoverflow.com/questions/19113725/what-dependency-between-font-size-and-width-of-char-in-monospace-font) ([here](https://web.archive.org/web/20010717031241/plainlanguagenetwork.org/type/utbo211.htm) is an archive of the mentioned original source) suggest sthat a 60 % width-to-height ratio may be a quite good approximation for monospace fonts.

So if I am not too wrong, a formula like:
```
Nmax of chars = width in inches / ((0.6 + epsilon) * fontsize in points * 1/72) 
```
may be a rather easy way to estimate the number of characters that should fit in a given width. (PS: in case it is not clear, the PostScript point is 1/72 of international inch)

Anyway, AFAIU @anntzer´s suggestion, his idea was more to use columns of minimal width and that do not wrap the text (which may be easier to implement ^^)